hydra-node/test/Hydra/BehaviorSpec.hs

```diff
@@ -613,13 +615,15 @@ simulatedChainAndNetwork initialChainState = do
     now <- getCurrentTime
     event <- atomically $ do
       cs <- readTVar chainStateVar
-      pure $ Tick now (chainStateSlot cs)
+      pure $ Tick now (chainStateSlot (currentState cs))
     readTVarIO nodes >>= mapM_ (`handleChainEvent` event)
 
   createAndYieldEvent nodes history chainStateVar tx = do
     chainEvent <- atomically $ do
-      modifyTVar' chainStateVar advanceSlot
-      cs' <- readTVar chainStateVar
+      cs <- currentState <$> readTVar chainStateVar
+      let cs' = advanceSlot cs
+      modifyTVar' chainStateVar $ \(UnsafeChainStateHistory prev) ->
+        UnsafeChainStateHistory (cs' :| tail prev)
```

Comment:
As we have a full chain state history here, I think we should just `pushNewState` here.

Reply:
Thinking about this further, I wonder why we don't just use the `LocalChainState m` handle here as well? (That just packs a history into a tvar).

EDIT: We can't do that right away because `LocalChainState` is fixed for `Tx` and this code works on a generic `tx`. However, if we change the interface of `LocalChainState.rollback` to be using the generic `ChainSlot` as argument, we could stay abstract and use it here as well.

Reply:
Changes made:
- rollback is now using the generic `ChainSlot` as argument
- LocalChainState is now abstract over tx